### DISCUSSION TITLE
What are the downsides to proxying a database connection?

### DISCUSSION THREAD

**Discussion by alexklibisz** [linkerd/linkerd2#5415] [Q&A] [answered]
I just realized that one of our service deployments was proxying JDBC connections from a JVM app to a postgres database. This is a pretty performance-critical connection, so I disabled it, figuring there must be some overhead to proxying a postgres connection (i.e. "no free lunch"). Now I'm wondering what could be the concrete side-effects or disadvantages to this?

**Comment by adleong** [comment, marked as answer]
The overhead to proxying connections is generally pretty small, but I recommend measuring the difference yourself to see what the overhead is in your particular scenario and decide if that overhead is acceptable or not.  By skipping the proxy for these connections, you will lose all the Linkerd functionality for those connections such as mTLS and TCP metrics.

**Reply by alexklibisz** [reply]
Thanks. For now we've removed the proxy on port 5432 just for simplicity's sake.
